### PR TITLE
Refactor repository

### PR DESCRIPTION
**Pull request by bedroesb**
Main differences are:

- Simpler structure reflecting the latest decisions, including the creation of a Python library that can be used by a data broker platform of choice. (this could ARC, Galaxy or a webapp to be made)
- Highlighting the difference between the Data broker - Data broker platform - MARS-CLI
- Introduction of the MARS-CLI backbone based on code written by Marcos (biosamples_external_references.py) (Thanks @kdp-cloud )
- Taking out old links and files from BioHackathon
- Hosting the repository services in one folder called `repository-services` including `isajson-biosamples` and  `isajson-ena`. Accompanied with a docker compose and a README to set them up
- README explaining some components of MARS
  - ISA-JSON as metadata carrier
  - Data broker platform
  - MARS-CLI
  - ISA-JSON support by repositories
- Test data bundled together in the `test-data` folder
- Description of the stakeholders updated and moved to a separate file
- Taking out the code for the JS webapp, since currently not compatible with the Python Library route we decided to take. Can be brought back in if possible.
- New diagram

**[PREVIEW the repo structure here](https://github.com/elixir-europe/MARS/tree/refactor)**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,40 +1,131 @@
...
 
-During this BioHackathon project, we aim to advance the development of the approach defined last year, by including other ELIXIR and non-ELIXIR repositories, namely MetaboLights, BioStudies/ArrayExpress and e!DAL-PGP, to cover multi-omics submission. 
+## Stakeholders
 
-To reinforce the broad applicability of the approach, we will work with ISA-JSON files generated by the ARC data structure, in addition to the ones generated by DataHub. We will also further explore the application of stepwise validated data flows orchestrated through Omnipy to extend the existing prototype of the converter tool. Defining requirements for the implementation of the independent upload tool is also in scope to support the feasibility of a streamlined brokering process.
+MARS is comprised of multiple stakeholders, the end-user, the platform that generates the ISA-JSON, target repositories and the data broker. Each representing key roles in the data submission process. Read more about it in our [stakeholders page](/stakeholders.md).
```
Comment (apriltuesday) [inline README.md]:
```suggestion
MARS is comprised of multiple stakeholders: the end-user, the platform that generates the ISA-JSON, target repositories and the data broker. Each represents key roles in the data submission process. Read more about it in our [stakeholders page](/stakeholders.md).
```

**Review by apriltuesday — APPROVED** [review]
The code organisation and docs make sense to me, thanks!
This might have already been discussed, but I guess we should confirm with @xiaoranzhou that there's a new repo for the webapp/GUI he's working on, so we can at least link to it as an example implementation.

**Comment by bedroesb** [comment]
@apriltuesday I agree. I think it would make sense that the webapp from @xiaoranzhou would be part of an implementation example of MARS using ARC. We could improve on the implementation in the Data Broker platform section of the README.

**Comment by xiaoranzhou** [comment]
Hi, I have just created a pull request to the BioHackathon 27 repo by using the following code.
https://github.com/xiaoranzhou/mars/tree/27-web
Yesterday DataPLANT had a meeting and we are going to integrate the web app into our [PlantDataHub](https://git.nfdi4plants.org/explore). More developments are needed for a production ready tool.

**Comment by floradanna** [comment]
@xiaoranzhou it is great to hear that PlantDataHub will use the web app. It would be nice if the web app would use the python library we are making for MARS. If you need any help in making use of the MARS python library let us know. 

In Belgium, we plan to deploy the MARS-CLI in Galaxy Belgium (as Galaxy tool).

I would suggest to keep the code of each deployment of MARS separated in a different repository.

**Review by kdp-cloud — APPROVED** [review]

**Comment by xiaoranzhou** [comment]
> @xiaoranzhou it is great to hear that PlantDataHub will use the web app. It would be nice if the web app would use the python library we are making for MARS. If you need any help in making use of the MARS python library let us know.
> 
> In Belgium, we plan to deploy the MARS-CLI in Galaxy Belgium (as Galaxy tool).
> 
> I would suggest to keep the code of each deployment of MARS separated in a different repository.

Yes, we plan to use the CLI for sure. I am also participating the development and testing. So there won't be any problem. 
Should we use the same MARS-CLI or have some adaptation of each implementation? 
If DataPLANT is have own implementation, it should be also listed in the MARS project.

**Comment by bedroesb** [comment]
hello @xiaoranzhou. I think it would be the most beneficial if it is the same code we are using for the same functions. Otherwise when for example the ISA-JSON improves/changes in shape, we have to update a lot more code. In your implementation it might be nice to have a flask/fastapi instance that makes use of the MARS-CLI as a python library, where you could communicate with. 

Sorry if I understand things wrongly!

**Comment by xiaoranzhou** [comment]
> hello @xiaoranzhou. I think it would be the most beneficial if it is the same code we are using for the same functions. Otherwise when for example the ISA-JSON improves/changes in shape, we have to update a lot more code. In your implementation it might be nice to have a flask/fastapi instance that makes use of the MARS-CLI as a python library, where you could communicate with.
> 
> Sorry if I understand things wrongly!

- Same code: yes, I agree. 
- Implementation: DataPLANT also use CLI tools and would like to do it in the CI/CD of GitLab. I think it will work for any CLI tools.

**Comment by bedroesb** [comment]
@xiaoranzhou that sounds great! It needs for sure to be clear from this repo which types of implementations and solutions exist.

**Pull request merged**